packages/procaptcha-bundle/src/index.tsx

```diff
@@ -114,26 +77,47 @@ export default function ready(fn: () => void) {
 // extend the global Window interface to include the procaptcha object
 declare global {
 	interface Window {
-		procaptcha: { ready: typeof ready; render: typeof render };
+		procaptcha: {
+			ready: typeof ready;
+			render: typeof render;
+			reset: typeof reset;
+		};
 	}
 }
 
-// set the procaptcha attribute on the window
-window.procaptcha = { ready, render };
+const start = () => {
+	// onLoadUrlCallback defines the name of the callback function to be called when the script is loaded
+	// onRenderExplicit takes values of either explicit or implicit
+	const { onloadUrlCallback, renderExplicit } = extractParams(BUNDLE_NAME);
 
-// onLoadUrlCallback defines the name of the callback function to be called when the script is loaded
-// onRenderExplicit takes values of either explicit or implicit
-const { onloadUrlCallback, renderExplicit } = extractParams(BUNDLE_NAME);
+	// Render the Procaptcha component implicitly if renderExplicit is not set to explicit
+	if (renderExplicit !== "explicit") {
+		ready(implicitRender);
+	}
 
-// Render the Procaptcha component implicitly if renderExplicit is not set to explicit
-if (renderExplicit !== "explicit") {
-	ready(implicitRender);
-}
+	if (onloadUrlCallback) {
+		const onloadCallback = getWindowCallback(onloadUrlCallback);
+		// Add event listener to the script tag to call the callback function when the script is loaded
+		getProcaptchaScript(BUNDLE_NAME)?.addEventListener("load", () => {
+			ready(onloadCallback);
+		});
+		// or if the document has already loaded, call the callback function
+		if (document.readyState === "complete") {
+			ready(onloadCallback);
+		}
```

Comment:
Can this cause ready() to be called twice?

Reply:
Added a check.